Review the Swan Lake Preview 2 Release Note (#1071)
ballerina-platform/ballerina-dev-website: downloads/swan-lake-release-notes/swan-lake-preview2/RELEASE_NOTE.md

}
```

Since `details` is created as an immutable value, the following check evaluates to `true`.

```ballerina
details.isReadOnly()
```

> ### The rest argument to pass arguments for required/defaultable parameters

Reviewer: This doesn't quite sound right. @MaryamZi thoughts please

Author: Verbs or full sentences should not be used in titles.

Reviewer 2: Yeah, I don't think this conveys the idea. Can we please change this to something like 

"Passing arguments for required/defaultable parameters via the rest argument"

Author: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/1072.

### HTTP

The redirect client has been improved to support temporary and permanent redirects with the original request’s HTTP method. A new `allowAuthHeaders` field was introduced in the redirect configuration to get user consent before repeating sensitive data on subsequent requests.

### Log

> A `log:setModuleLogLevel` function has been introduced to allow users to set the module log level through the Log API.

Reviewer: ```suggestion
The new `log:setModuleLogLevel` function allows users to set the module log level through the Log API.
```

Author: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/1072.

### Config

The `config:getAsArray` and `config:getAsMap` functions now return immutable values. This way, the user can simply cast the returned value to the desired list type or mapping type without having to use `cloneWithType()`.

### Security

- JWT signature validation with JWK has been improved by introducing a cache. This cache preloads the JWKs, and thereby, reduces the runtime latency added due to the HTTP call to the JWK endpoint when validating the signature.
- An `auth:InvocationContext` record, which can be used as a data holder has been introduced. It can hold auth-related information such as authentication scheme, auth token, and authenticated user's ID, claims, and scopes preserved for a single request-response flow.

### WebSocket

> The WebSocket client now supports cookies. A new field `cookies` has been introduced in the `http:WebSocketClientConfiguration` and `http:WebSocketFailoverClientConfiguration` to specify the cookies.

Reviewer 2: If we are adding "the" shouldn't this be

```suggestion
The WebSocket client now supports cookies. A new field `cookies` has been introduced in the `http:WebSocketClientConfiguration` and `http:WebSocketFailoverClientConfiguration` records to specify the cookies.
```

`http:WebSocketClientConfiguration` and `http:WebSocketFailoverClientConfiguration` are names.

Author: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/1072.

### Runtime API

- The `runtime:timeout` function has been removed. The `runtime:sleep` function can be used as the alternative.

> For example, when the following, which previously used the `runtime:timeout` is considered,

Reviewer 2: This starts here and ends at L260. Feels like an incomplete sentence to me now.

Can we please change this also to something like

"Consider the following example."

and then change L257 to

"This can now be done as follows:"

Author: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/1072.

> 

Reviewer: For the above, how about just captioning the two code snippets as "Previous method" and "New method" ?

Reviewer 2: `sleep` isn't new right? So we can't really say new method.

Reviewer: Is it misleading? I meant method as in way. Not method as in function. :D

Reviewer 2: At least for me it was :D Can we not just do https://github.com/ballerina-platform/ballerina-dev-website/pull/1071#discussion_r456560475

Author: +1 Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/1072.
